spirv_msl.cpp

```diff
@@ -1253,7 +1253,16 @@ string CompilerMSL::entry_point_args(bool append_comma)
 				case SPIRType::Struct:
 					if (!ep_args.empty())
 						ep_args += ", ";
-					ep_args += "constant " + type_to_glsl(type) + "& " + to_name(var.self);
+					if ((meta[type.self].decoration.decoration_flags & (1ull << DecorationBufferBlock)) != 0 ||
+						(meta[type.self].decoration.decoration_flags & (1ull << DecorationNonWritable)) == 0)
```

Comment:
The test should be more in tune of
if (BufferBlock && NonWritable == 0) device; else constant;

Regular UBOs are never writable, the decoration is kind of meaningless for them.

I'm not sure about the semantics for device vs. constant however in Metal. Does constant mean readonly, or does it imply that the buffer is found in special constant memory (UBO)?